src/DefaultBuilder/test/Microsoft.AspNetCore.TrimmingTests/Microsoft.AspNetCore.TrimmingTests.proj

```diff
@@ -0,0 +1,14 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <ItemGroup>
+    <TestConsoleAppSourceFiles Include="SlimBuilderDoesNotDependOnX509Test.cs">
+      <DisabledFeatureSwitches>System.Text.Json.JsonSerializer.IsReflectionEnabledByDefault</DisabledFeatureSwitches>
```

Comment:
What was happening when you didn't set this? You were getting trim warnings during publish and that broke your test?

Reply:
Note that the `NativeAotTests.proj` doesn't need to do this because `IsReflectionEnabledByDefault` is defaulted to `false` when `PublishAot=true`. https://github.com/dotnet/runtime/issues/84378 is tracking doing it for `PublishTrimmed` as well.

Reply:
When it lived in NativeAotTests, I was getting a warning about it.  In TrimmingTests, it was an error, which broke build.

Reply:
For posterity, the error was
```
src\Http\Http.Extensions\src\JsonOptions.cs(39,12): Trim analysis error IL2026: Microsoft.AspNetCore.Http.Json.JsonOptions.CreateDefaultTypeResolver(): Using member 'System.Text.Json.Serialization.Metadata.DefaultJsonTypeInfoResolver.DefaultJsonTypeInfoResolver()' which has 'RequiresUnreferencedCodeAttribute' can break functionality when trimming application code. JSON serialization and deserialization might require types that cannot be statically analyzed. Use the overload that takes a JsonTypeInfo or JsonSerializerContext, or make sure all of the required types are preserved.
```